src/Sarif.WorkItemFiling/FilingClientFactory.cs

```diff
@@ -7,25 +7,36 @@
 
 namespace Microsoft.CodeAnalysis.Sarif.WorkItemFiling
 {
-    public class FilingTargetFactory
+    public class FilingClientFactory
     {
-        private const string AzureDevOpsUriPattern = @"^https://dev.azure.com/[^/]+/[^/]+$";
+        private const string LegacyAzureDevOpsUriPattern = @"^https://[^\.]+\.visualstudio\.com/[^/]+$";
+        private static readonly Regex s_legacyAzureDevOpsUriRegex = SarifUtilities.RegexFromPattern(LegacyAzureDevOpsUriPattern);
+
+        private const string AzureDevOpsUriPattern = @"^https://dev\.azure\.com/[^/]+/[^/]+$";
         private static readonly Regex s_azureDevOpsUriRegex = SarifUtilities.RegexFromPattern(AzureDevOpsUriPattern);
 
-        private const string GitHubUriPattern = @"^https://github.com/[^/]+/[^/]+$";
+        private const string GitHubUriPattern = @"^https://github\.com/[^/]+/[^/]+$";
         private static readonly Regex s_gitHubUriRegex = SarifUtilities.RegexFromPattern(GitHubUriPattern);
 
-        public static FilingTarget CreateFilingTarget(string projectUriString)
+
+        // https://secretscantest.visualstudio.com/RTAKET-DELM
+
+        public static FilingClient CreateFilingTarget(string projectUriString)
         {
             if (projectUriString == null) { throw new ArgumentNullException(nameof(projectUriString)); }
 
+            // TODO: make this code populate project/account/org details 
             if (s_azureDevOpsUriRegex.IsMatch(projectUriString))
             {
-                return new AzureDevOpsFilingTarget();
+                return new AzureDevOpsClient();
+            }
+            else if (s_legacyAzureDevOpsUriRegex.IsMatch(projectUriString))
+            {
+                return new AzureDevOpsClient();
             }
             else if (s_gitHubUriRegex.IsMatch(projectUriString))
             {
-                return new GitHubFilingTarget();
+                return new GitHubClient();
             }
             else
             {
```

Comment:
I suspect your most commonly in this state because the user left off the project portion of the URI.  If we don't take Lila's suggestion above of having the user specify them explicitly, I would definitely re-iterate the format expect in this exception, even if it's already in the help text.

Reply:
Agreed, will put the hood up here in the next check-in and do what you suggest.

---
In reply to: [375988292](https://github.com/microsoft/sarif-sdk/pull/1744#discussion_r375988292) [](ancestors = 375988292)